weave/compile.py

```diff
@@ -87,6 +87,39 @@ def _quote_node(node: graph.Node) -> graph.Node:
     return weave_internal.const(node)
 
 
+def _node_simplification(node: graph.Node) -> typing.Optional[graph.OutputNode]:
+    if isinstance(node, graph.OutputNode) and node.from_op.name == "count":
+        arr_node = node.from_op.inputs["arr"]
+        if isinstance(arr_node, graph.OutputNode) and arr_node.from_op.name.startswith(
+            "run-history"
+        ):
+            run_node = arr_node.from_op.inputs["run"]
+            return graph.OutputNode(
+                node.type,
+                "run-historyLineCount",
+                {"run": run_node},
+            )
+    elif isinstance(node, graph.OutputNode) and node.from_op.name == "unique":
```

Comment:
This seems to specific. Can you make it work for regular list operations too?

Reply:
This is actually pretty hard. if we have a list[dict], the resolver would still need to iterate through all the elements to determine the keys. We benefit from AWL because the type info comes along for the ride into the value space. I'll think more on it

Reply:
Yup definitely have to iterate. So maybe you're saying it doesn't matter in that case though, there's no perf gain. That's fair.

Reply:
yeah exactly. the only way we can get it to be a perf gain is if we can effectively leverage the node type (which should already have processed the rows)